tmt/steps/execute/__init__.py

```diff
@@ -562,31 +562,48 @@ def load_tmt_report_results(self, invocation: TestInvocation) -> list["tmt.Resul
             return []
 
         # Check the test result
-        self.debug(f"tmt-report-results file '{report_result_path} detected.")
+        self.debug(f"tmt-report-results file '{report_result_path}' detected.")
+
+        result_list = [
+            result.get('result')
+            for result in tmt.utils.yaml_to_list(self.read(report_result_path))
+            ]
```

Comment:
* since we're using "outcome" and `ResultOutcome` enum for "pass", "error" and other "outcomes", I'd propose using this name to dilute the amount of "result" labels here and in the following lines.
* I would also convert the string to their enum representation, so we can catch possible misuse in code as soon as possible - internally, `ResultOutcome` is *the* type to use when representing result outcomes, and this is where input enters tmt internals, therefore the conversion should happen. And since you're in the area, I'd say it's a good opportunity to sort out the naming scheme in lines you're already changing.
* using `_list` suffix to announce the variable is a list is not as common as using the plural, i.e. `results` instead of `result_list`.
* similarly, `None` is way better than an empty string when it comes to "unset" or "undefined" value. It's trivial for a linter to catch the situation where a variable is used despite being unset when `None` is the marker, on the other hand, an empty string is indistinguishable from a non-empty string.

All in all, I came up with something like this - sorry for rewriting the code for you, but once I started digging, it turned out the naming changes and moving code around to simplify it would result in very useless per-line comments :/

```python
        # Check the test result
        self.debug(f"tmt-report-results file '{report_result_path}' detected.")

        # The worst result outcome we can find among loaded results...
        original_outcome: Optional[ResultOutcome] = None
        # ... and the actual outcome we decided is the best representing
        # the results.
        # The original one may be left unset - malformed results file,
        # for example, provides no usable original outcome.
        actual_outcome: ResultOutcome
        note: Optional[str] = None

        results = tmt.utils.yaml_to_list(self.read(report_result_path))

        if not results:
            raise tmt.utils.ExecuteError(
                f"Test result not found in result file '{report_result_path}'.")

        try:
            outcomes = [
                ResultOutcome.from_spec(result.get('result'))
                for result in results
            ]

        except tmt.utils.SpecificationError as exc:
            actual_outcome = ResultOutcome.ERROR
            note = exc.message

        else:
            hierarchy = [ResultOutcome.SKIP, ResultOutcome.PASS, ResultOutcome.WARN, ResultOutcome.FAIL]
            outcome_indices = [hierarchy.index(outcome) for outcome in outcomes]
            actual_outcome = original_outcome = hierarchy[max(outcome_indices)]

        if actual_outcome is ResultOutcome.SKIP:
            actual_outcome = ResultOutcome.INFO

        # Find a usable log - the first one matching our "interim" outcome.
        # We cannot use the "actual" outcome, because that one may not even
        # exist in the results file - tmt might have conjured it based on
        # provided results, or set it to "error" because of errors. Only
        # the "interim" is guaranteed to be found among the results.
        test_logs = [invocation.relative_path / TEST_OUTPUT_FILENAME]

        if original_outcome is not None:
            for result in results:
                if result.get('result') != original_outcome.value:
                    continue

                result_logs: list[str] = log.get('log', [])

                if result_logs:
                    test_logs.append(invocation.relative_test_data_path / result_logs[0])

                break
        
        return [tmt.Result.from_test_invocation(
            invocation=invocation,
            result=actual_result,
            log=test_logs,
            note=note)]
```

What do you think?

Reply:
Indeed, that looks tidier. I've included it in my most recent push.

I removed this section as it's desired to have the result of 'skip' returned rather than altering it to 'info'
```
        if actual_outcome is ResultOutcome.SKIP:
            actual_outcome = ResultOutcome.INFO
```
There were also a couple of used variables which were undeclared, so I corrected that. Otherwise, it works well.